core/src/services/opfs/backend.rs

```diff
@@ -0,0 +1,240 @@
+use serde::Deserialize;
+use std::sync::Arc;
+
+use crate::{
+    raw::{
+        Access, AccessorInfo, OpCopy, OpCreateDir, OpDelete, OpList, OpRead, OpRename, OpStat,
+        OpWrite, RpCopy, RpCreateDir, RpDelete, RpList, RpRead, RpRename, RpStat, RpWrite,
+    },
+    types, Builder, Capability, Error, Result, Scheme,
+};
+use std::fmt::Debug;
+
+use super::{core::OpfsCore, lister::OpfsLister, reader::OpfsReader, writer::OpfsWriter};
+
+/// Origin private file system (OPFS) configuration
+#[derive(Default, Deserialize)]
+#[serde(default)]
+#[non_exhaustive]
+pub struct OpfsConfig {}
+
+impl Debug for OpfsConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        panic!()
+    }
+}
+
+/// Origin private file system (OPFS) support
+#[doc = include_str!("docs.md")]
+#[derive(Default)]
+pub struct OpfsBuilder {
+    config: OpfsConfig,
+}
+
+impl OpfsBuilder {}
+
+impl Debug for OpfsBuilder {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        panic!()
+    }
+}
+
+impl Builder for OpfsBuilder {
+    const SCHEME: Scheme = Scheme::Opfs;
+
+    type Config = ();
+
+    fn build(self) -> Result<impl Access> {
+        Ok(OpfsBackend {})
+    }
+}
+
+/// OPFS Service backend
+#[derive(Debug, Clone)]
+pub struct OpfsBackend {}
+
+impl Access for OpfsBackend {
+    type Reader = OpfsReader;
+
+    type Writer = OpfsWriter;
+
+    type Lister = OpfsLister;
+
+    type BlockingLister = OpfsLister;
+
+    type BlockingReader = OpfsReader;
+
+    type BlockingWriter = OpfsWriter;
+
+    fn info(&self) -> Arc<AccessorInfo> {
+        let mut access_info = AccessorInfo::default();
+        access_info
+            .set_scheme(Scheme::Opfs)
+            .set_native_capability(Capability {
+                stat: true,
+                read: true,
+                write: true,
+                write_can_empty: true,
+                write_can_append: true,
+                write_can_multi: true,
+                create_dir: true,
+                delete: true,
+                list: true,
+                copy: true,
+                rename: true,
+                blocking: true,
+                ..Default::default()
+            });
+        Arc::new(access_info)
+    }
+
+    async fn create_dir(&self, path: &str, _: OpCreateDir) -> Result<RpCreateDir> {
```

Comment:
We have default implementations for those functions, so we can remove unsupported operations.